lisa/schema.py

```diff
@@ -1309,6 +1310,67 @@ def type_name(cls) -> str:
         return constants.TESTCASE_TYPE_LEGACY
 
 
+@dataclass_json()
+@dataclass
+class ConnectionInfo:
+    address: str = ""
+    port: int = field(
+        default=22,
+        metadata=field_metadata(
+            field_function=fields.Int, validate=validate.Range(min=1, max=65535)
+        ),
+    )
+    username: str = constants.DEFAULT_USER_NAME
+    password: Optional[str] = ""
+    private_key_file: Optional[str] = ""
+
+    def __post_init__(self, *args: Any, **kwargs: Any) -> None:
+        add_secret(self.username, PATTERN_HEADTAIL)
+        add_secret(self.password)
+        add_secret(self.private_key_file)
+
+        if not self.password and not self.private_key_file:
+            raise LisaException(
+                "at least one of password or private_key_file need to be set when "
+                "connecting"
+            )
+        elif not self.private_key_file:
```

Comment:
Would this disallow use of an empty password? It's terrible practice to use empty passwords and I am onboard with just not supporting that. If it's an undocumented requirement that's a side effect of an internal library we should make sure to warn at runtime and also document it.